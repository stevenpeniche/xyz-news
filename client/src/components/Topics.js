import React from 'react';
import './Topics.css';

const Topics = (props) => {
  return (
    <div className="source-topics-container">
      <div className="source-topics animated fadeIn">
        <div className="topics-header">
          Topics
        </div>
        <ul className="topics-list">
          {props.topics.map((topic) => {
            return (
              <li
                className={topic.name === props.currentTopic ? "topic active-topic" : "topic"}
                key={topic.name}
                onClick={() => props.changeCurrentTopic(topic.name)}
              >
                {topic.name}
              </li>
            )
          })}
        </ul>
      </div>
    </div>
  );
};

export default Topics;