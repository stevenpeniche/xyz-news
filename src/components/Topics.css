.topics-header {
  font-size: 18px;
  color: #797979;
}

.topics-list {
  list-style-type: none;
  padding-left: 16px;
}

.topic {
  margin-top: 15px;
  margin-bottom: 15px;
  transition-duration: 400ms;
}

.topic:hover {
  cursor: pointer;
  color: #ff6600;
}

.active-topic {
  color: #ff6600;
}