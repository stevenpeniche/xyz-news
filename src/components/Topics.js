import React from 'react';
import shortid from 'shortid';
import './Topics.css';

const Topics = (props) => {
  return (
    <div className="source-topics-container">
      <div className="source-topics">
        <div className="topics-header">
          Topics
        </div>
        <ul className="topics-list">
          {props.topics.map((topic) => {
            return (
              <li
                className={topic === props.currentTopic ? "topic active-topic" : "topic"}
                key={shortid.generate()}
                onClick={() => props.changeCurrentTopic(topic)}
              >
                {topic}
              </li>
            )
          })}
        </ul>
      </div>
    </div>
  );
};

export default Topics;