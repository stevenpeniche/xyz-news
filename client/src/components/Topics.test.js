import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Topics from './Topics';

let currentTopic, source, changeCurrentTopic, wrapper;

beforeAll(() => {
  currentTopic = 'Top';

  source = {
    id: 'bleacher-report',
    primaryColor: '#000000',
    topics: [
      {
        name: 'Top'
      },
      {
        name: 'MLB'
      },
      {
        name: 'NFL'
      },
      {
        name: 'NBA'
      },
      {
        name: 'UFC'
      },
      {
        name: 'Boxing'
      },
      {
        name: 'WWE'
      }
    ]
  };

  changeCurrentTopic = jest.fn();

  wrapper =
    shallow(
      <Topics
        sourceColor={source.primaryColor}
        topics={source.topics}
        currentTopic={currentTopic}
        changeCurrentTopic={changeCurrentTopic}
      />
    );
});

describe('App rendering', () => {
  it('renders without crashing', () => {
    wrapper
  });

  it('renders correctly', () => {
    const tree =
      renderer.create(
        <Topics
          sourceColor={source.primaryColor}
          topics={source.topics}
          currentTopic={currentTopic}
          changeCurrentTopic={changeCurrentTopic}
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the proper number of list elements", () => {
    expect(wrapper.find('li').length).toEqual(source.topics.length);
  });
});
