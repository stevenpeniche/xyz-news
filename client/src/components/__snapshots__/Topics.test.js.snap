// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topics rendering renders correctly 1`] = `
<div
  className="source-topics-container"
>
  <div
    className="source-topics animated fadeIn"
  >
    <div
      className="topics-header"
    >
      Topics
    </div>
    <ul
      className="topics-list"
    >
      <li
        className="topic active-topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#A9A9A9",
          }
        }
      >
        Top
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        MLB
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        NFL
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        NBA
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        UFC
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        Boxing
      </li>
      <li
        className="topic"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={Object {}}
      >
        WWE
      </li>
    </ul>
  </div>
</div>
`;
